### PR TITLE
feat: Spin support percent

### PR DESCRIPTION
**Pull request by zombieJ**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref #48647

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Spin support `percent` to render as progress.      |
| 🇨🇳 Chinese |    Spin 支持 `percent`属性以进度形态展示。     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
感觉可以搞一个 nprogress 模式了

File: components/spin/Indicator/Progress.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  // ==================== Progress ====================
+  const safePtg = Math.max(Math.min(percent, 100), 0);
+
+  const viewSize = 100;
+  const borderWidth = viewSize / 5;
+  const radius = viewSize / 2 - borderWidth / 2;
+  const circumference = radius * 2 * Math.PI;
+
+  const renderCircle = (circleClassName?: string, style?: React.CSSProperties) => (
```
Comment (li-jia-nan) [inline components/spin/Indicator/Progress.tsx:32]: 这两个参数感觉用对象好一点，就不用关心入参顺序了

Comment (zombieJ) [reply components/spin/Indicator/Progress.tsx:32]: 对象会增加 bundle size，参数本身不对切是内部方法，打平足够了~

**Review by yoyo837 — DISMISSED** [review]

File: components/spin/Indicator/Looper.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const dotClassName = `${prefixCls}-dot`;
+  const holderClassName = `${dotClassName}-holder`;
+  const hideClassName = `${holderClassName}-hidden`;
+
+  // ===================== Render =====================
+  return (
+    <>
+      <span className={classNames(holderClassName, percent > 0 && hideClassName)}>
+        <span className={classNames(dotClassName, `${prefixCls}-dot-spin`)}>
+          <i className={`${prefixCls}-dot-item`} key={1} />
```
Comment (crazyair) [inline components/spin/Indicator/Looper.tsx:22]: map?

File: components/spin/Indicator/Progress.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  if (!render) {
+    return null;
+  }
+
+  return (
+    <span
+      className={classNames(
+        holderClassName,
+        `${dotClassName}-progress`,
+        safePtg <= 0 && hideClassName,
```
Comment (crazyair) [inline components/spin/Indicator/Progress.tsx:53]: `[xxx]: boolean`

File: components/spin/Indicator/Looper.tsx (hunk 2 above)
Comment (zombieJ) [reply components/spin/Indicator/Looper.tsx:22]: 老代码，没动。可以改

**Review by crazyair — DISMISSED** [review]

**Review by crazyair — DISMISSED** [review]

File: components/spin/Indicator/Progress.tsx (hunk 3 above)
Comment (zombieJ) [reply components/spin/Indicator/Progress.tsx:53]: 这里没有 object，加了会增加 size 所以省了一下下~~

**Review by li-jia-nan — DISMISSED** [review]

**Comment by afc163** [comment]
覆盖率掉了。

**Review by li-jia-nan — DISMISSED** [review]

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**